### PR TITLE
Fix TS errors in contributionsLandingActions.ts & associated files

### PR DESCRIPTION
**Pull request by rupertbates**
## What are you doing in this PR?
This is preparation for some work I'm doing in this area, I wanted to start from error free files.

## Is this an AB test?
- [ ] Yes
- [x] No

If this is an AB test, PR reviewers should open and check the Optimize test.
[**Optimize Link**](https://optimize.google.com/optimize/home)

## Accessibility test checklist
 - [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
 - [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
 - [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### CODE REVIEW

File: support-frontend/assets/helpers/forms/paymentMethods.ts [hunk 1]
```diff
@@ -17,12 +17,17 @@ export type PaymentMethodMap<T> = {
...
 	AmazonPay: T;
 };
-// This lets us create a union type from the object keys,
-// avoiding the need to specify them separately and keep them in sync!
-// https://flow.org/en/docs/types/utilities/#toc-keys
-// We need to supply the type parameter, but we're only using the keys
-// so it's irrelevant - so we supply null
-export type PaymentMethod = $Keys<PaymentMethodMap<null>>;
+
+export type PaymentMethod =
```
Comment (tomrf1) [inline support-frontend/assets/helpers/forms/paymentMethods.ts:21]: could this be simpler now?
`export type PaymentMethod = 'Stripe' | 'PayPal' | ...`

Comment (tomrf1) [reply support-frontend/assets/helpers/forms/paymentMethods.ts:21]: actually, I'm guessing there are loads of places that import those consts, which would make it a very big PR...

**Review by tomrf1 — APPROVED** [review]
looks good!

File: support-frontend/assets/helpers/forms/paymentMethods.ts (hunk 1 above)
Comment (rupertbates) [reply support-frontend/assets/helpers/forms/paymentMethods.ts:21]: Yes that's the reason I didn't get rid of `PaymentMethodMap` as well even though I don't think we need it now - it was basically working round some limitations of Flow. 
Let's save those for another PR!

File: support-frontend/assets/helpers/subscriptionsForms/submit.ts [hunk 2]
```diff
@@ -110,14 +114,14 @@ const getProduct = (
...
-		currency: currencyId || state.common.internationalisation.currencyId,
+		currency: currencyId ?? state.common.internationalisation.currencyId,
 		billingPeriod,
 		fulfilmentOptions: fulfilmentOption,
 		productOptions: productOption,
 	};
 };
 
-const getPromoCode = (promotions: Promotion[] | null | undefined) => {
+const getPromoCode = (promotions: Promotion[] | undefined) => {
```
Comment (GHaberis) [inline support-frontend/assets/helpers/subscriptionsForms/submit.ts:124]: Could we make the `promotions` argument optional and get rid of the `| undefined`?

```
const getPromoCode = (promotions?: Promotion[]) => { ... }
```

**Review by GHaberis — APPROVED** [review]
Looks good!

**Pull request merged**